core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -198,9 +205,11 @@ public ChannelImpl shutdown() {
    *
    * <p>NOT YET IMPLEMENTED. This method currently behaves identically to shutdown().
    */
-  // TODO(ejona86): cancel preexisting calls.
   public ChannelImpl shutdownNow() {
     shutdown();
+    for (ClientCall<?, ?> call : activeCalls) {
+      call.cancel();
```

Comment:
We would need to change the API to make it thread-safe. Let's use the stream here and discuss separately whether it should be thread-safe.